components/Tabs/Tabs.js

```diff
@@ -1,51 +1,63 @@
-
 class TabLink {
   constructor(element) {
     // Assign this.element to the passed in DOM element
-    // this.element;
-    
+    this.element = element;
+
     // Get the custom data attribute on the Link
-    // this.data;
-    
+    this.data = this.element.dataset.tab;
+
     // Using the custom data attribute get the associated Item element
-    // this.itemElement;
-    
+    this.itemElement = document.querySelector(
+      `.tab-item[data-tab="${this.data}"]`
+    );
+
     // Using the Item element, create a new instance of the TabItem class
-    // this.tabItem;
-    
+    this.tabItem = new TabItem(this.itemElement);
+    //Why itemElement???
     // Add a click event listener on this instance, calling the select method on click
-
+    this.element.addEventListener('click', this.select(this));
   };
 
   select() {
     // Get all of the elements with the tabs-link class
-    // const links;
+    const links = document.querySelectorAll(".tabs-link");
+
+    // Using a loop or the forEach method remove the 'tabs-link-selected' class from all of the links
 
+    Array.from(links).forEach(tab =>
+      tab.classList.remove("tabs-link-selected")
+    );
+    // Need to learn how to do this from scratch... Need practice
     // Using a loop or the forEach method remove the 'tabs-link-selected' class from all of the links
-    // Array.from(links).forEach();
+    // Array.from(links).forEach(link => {
+    //   document.querySelectorAll('.tabs-link-selected').remove
+    // });
 
     // Add a class named "tabs-link-selected" to this link
-    // this.element;
-    
-    // Call the select method on the item associated with this link
+    this.element = tab.classList.add("tabs-link-selected");
 
+    // Call the select method on the item associated with this link
+    //NEED TO FIGURE THIS OUT>>> DON'T UNDERSTAND
+    this.tabItem.select(this.itemElement);
   }
```

Comment:
what's a little confusing here is that we have two methods -- both called select. this is calling the select on the second class you have outlined here because we are starting with this.tabItem (this.tabItem refers to the TabItem class ... so we are pulling the select from that class as opposed to the TabLink class).